### DISCUSSION TITLE
LR keeps growing after using CyclicLR scheduler.

### DISCUSSION THREAD

**Discussion by Minxiangliu** [Lightning-AI/pytorch-lightning#12722] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I want to add `CyclicLR` during training. I refer to this [docs](https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#configure-optimizers) and the [monai tutorials](https://github.com/Project-MONAI/tutorials/blob/master/modules/learning_rate.ipynb), but I see in the log that LR is not changing with `Cyclic`, but increasing continuously. What is wrong with my code?

Code:
```
class Net(pytorch_lightning.LightningModule):
    def __init__(self, *args, **kwargs):
        super().__init__()
   .....

    def prepare_data(self):
        self.train_ds, self.test_ds = getDataSet(self.stage)

    def train_dataloader(self):
        return ThreadDataLoader(self.train_ds, 
                                num_workers=0, 
                                batch_size=self.batch_size, 
                                shuffle=True,
                                drop_last=True)

    def configure_optimizers(self):
        steepest_lr = 0.005
        optimizer = torch.optim.Adam(self.model.parameters(), lr = steepest_lr*0.5)
        self.scheduler = torch.optim.lr_scheduler.CyclicLR(optimizer=optimizer,
                                                           base_lr=steepest_lr*0.5,
                                                           max_lr=steepest_lr*1.5,
                                                           step_size_up=len(self.train_dataloader())*8,
                                                           cycle_momentum=False)
        return [optimizer], [self.scheduler]

    def training_step(self, batch, batch_idx):
        .....
        self.log('LR', {'learning_rate' : self.scheduler.get_last_lr()[0]}, on_epoch=False)
        return ...

```
Log:
![image](https://user-images.githubusercontent.com/30144428/162862303-b44997a2-5a87-4aee-9842-3c38ae8930c3.png)
